modules/helpers.go

```diff
@@ -0,0 +1,72 @@
+package modules
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prebid/prebid-server/hooks"
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+func createModuleStageNamesCollection(modules map[string]interface{}) (map[string][]string, error) {
+	moduleStageNameCollector := make(map[string][]string)
+	var added bool
+
+	for id, hook := range modules {
+		if _, ok := hook.(hookstage.Entrypoint); ok {
+			added = true
+			stageName := hooks.StageEntrypoint.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if _, ok := hook.(hookstage.RawAuctionRequest); ok {
+			added = true
+			stageName := hooks.StageRawAuctionRequest.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if _, ok := hook.(hookstage.ProcessedAuctionRequest); ok {
+			added = true
+			stageName := hooks.StageProcessedAuctionRequest.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if _, ok := hook.(hookstage.BidderRequest); ok {
+			added = true
+			stageName := hooks.StageBidderRequest.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if _, ok := hook.(hookstage.RawBidderResponse); ok {
+			added = true
+			stageName := hooks.StageRawBidderResponse.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if _, ok := hook.(hookstage.AllProcessedBidResponses); ok {
+			added = true
+			stageName := hooks.StageAllProcessedBidResponses.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if _, ok := hook.(hookstage.AuctionResponse); ok {
+			added = true
+			stageName := hooks.StageAuctionResponse.String()
+			moduleStageNameCollector = addModuleStageName(moduleStageNameCollector, id, stageName)
+		}
+
+		if !added {
+			return nil, fmt.Errorf(`hook "%s" does not implement any supported hook interface`, id)
+		}
+	}
+
+	return moduleStageNameCollector, nil
+}
+
+func addModuleStageName(moduleStageNameCollector map[string][]string, id string, stage string) map[string][]string {
+	replacer := strings.NewReplacer(".", "_", "-", "_")
```

Comment:
Can you move the replacer to a top level package scoped variable to save on some allocations?

Reply:
Done